numpy/core/src/multiarray/descriptor.c

```diff
@@ -596,6 +621,7 @@ _convert_from_list(PyObject *obj, int align)
     }
     /* End ignore code.*/
     if (n == 0) {
+        _report_generic_error();
```

Comment:
This snuck into this patch - I'll take it out if I make another revision, but it'll be coming in the next PR anyway.